CRM/Utils/System/WordPress.php

```diff
@@ -1509,4 +1509,103 @@ public function shouldExitAfterFatal() {
     return apply_filters('civicrm_exit_after_fatal', $ret);
   }
 
+  /**
+   * Make sure clean URLs are properly set in settings file.
+   *
+   * @return CRM_Utils_Check_Message[]
+   */
+  public function checkCleanurls() {
+    $config = CRM_Core_Config::singleton();
+    $clean = 0;
+    if (defined('CIVICRM_CLEANURL')) {
+      $clean = CIVICRM_CLEANURL;
+    }
+    if ($clean == 1) {
+      //cleanURLs are enabled in CiviCRM, let's make sure the wordpress permalink settings and cache are actually correct by checking the first active contribution page
+      $contributionPages = \Civi\Api4\ContributionPage::get(FALSE)
+        ->addSelect('id')
+        ->addWhere('is_active', '=', TRUE)
+        ->setLimit(1)
+        ->execute();
+      if (count($contributionPages) > 0) {
+        $activePageId = $contributionPages[0]['id'];
+        $message = self::checkCleanPage('/contribute/transact/?reset=1&id=', $activePageId, $config);
+
+        return $message;
+      }
+      else {
+        //no active contribution pages, we can check an event page. This probably won't ever happen.
+        $eventPages = \Civi\Api4\Event::get(FALSE)
+          ->addSelect('id')
+          ->addWhere('is_active', '=', TRUE)
+          ->setLimit(1)
+          ->execute();
+        if (count($eventPages) > 0) {
+          $activePageId = $eventPages[0]['id'];
+          $message = self::checkCleanPage('/event/info/?reset=1&id=', $activePageId, $config);
+
+          return $message;
+        }
+        else {
+          //If there are no active event or contribution pages, we'll skip this check for now.
+
+          return [];
+        }
+      }
+    }
+    else {
+      //cleanURLs aren't enabled or aren't defined correctly in CiviCRM, admin should check civicrm.settings.php
+      $warning = ts('Clean URLs  are not enabled correctly in CiviCRM. This can lead to "valid id" errors for users registering for events or making donations. Check civicrm.settings.php and review <a %1>the documentation</a> for more information.', [1 => 'href="' . CRM_Utils_System::docURL2('sysadmin/integration/wordpress/clean-urls/', TRUE) . '"']);
```

Comment:
Thanks @kcristiano.

@elisseck For spaces I meant literally there are 2 space characters directly after the words Clean URLs - right here. 
Otherwise thanks for sticking with it! - I think we're good.

```suggestion
      $warning = ts('Clean URLs are not enabled correctly in CiviCRM. This can lead to "valid id" errors for users registering for events or making donations. Check civicrm.settings.php and review <a %1>the documentation</a> for more information.', [1 => 'href="' . CRM_Utils_System::docURL2('sysadmin/integration/wordpress/clean-urls/', TRUE) . '"']);
```

Reply:
Ohh I see. Thanks!

Reply:
Cool - can you squash commits and then we can give it the big green label.